src/elf/mod.rs

```diff
@@ -276,7 +276,13 @@ if_sylvan! {
                 if dyn_info.needed_count > 0 {
                     libraries = dynamic.get_libraries(&dynstrtab);
                 }
-                let num_syms = if dyn_info.syment == 0 { 0 } else { if dyn_info.strtab <= dyn_info.symtab { 0 } else { (dyn_info.strtab - dyn_info.symtab) / dyn_info.syment }};
+                let num_syms = if let Some(gnu_hash) = dyn_info.gnu_hash {
+                    gnu_hash_len(bytes, gnu_hash as usize, ctx)?
+                } else if let Some(hash) = dyn_info.hash {
+                    hash_len(bytes, hash as usize, header.e_machine, ctx)?
+                } else {
+                    0
```

Comment:
Ok, was going to ask; in case of a non-existent gnu/hash table, we effectively don't parse any symbols. I'm ok with this, but now I'm wondering (from a purely scientific curiosity):

1. Can dynamic libraries _not_ have a hash table
2. How does the dynamic linker compute dynsym size if no? I imagine they must have hash tables if they're dynamic libraries?

Reply:
1. No, dynamic libraries must have a hash table (SYSV or GNU_HASH). Nevertheless the table can be empty (c.f. https://github.com/lief-project/LIEF/blob/master/src/ELF/Builder.cpp#L95-L133) 
   Object file (.o) don't have hash table, neither dynamic symbols.
2. Actually it doesn't compute the number of symbols. It look for an index and check that this index is consistent

Reply:
@romainthomas awesome, thanks for the clarification! It's super annoying to me the _DYNAMIC doesn't provide a `DT_SYMTABSZ`, like it does for basically everything else:

```
   DT_INIT_ARRAY 0x3d9060
 DT_INIT_ARRAYSZ 0x8
   DT_FINI_ARRAY 0x3d9068
 DT_FINI_ARRAYSZ 0x8
     DT_GNU_HASH 0x413548
       DT_STRTAB 0x415210
       DT_SYMTAB 0xc90
        DT_STRSZ 0x2554
       DT_SYMENT 0x18
```

Reply:
thanks :) if you ever felt like swapping out a backend, I'd definitely be interested in spec-ing out/ designing some C bindings for goblin (though it will probably be a lot of work?)